### DISCUSSION TITLE
Oprinal paramaters for createDraftListing

### DISCUSSION THREAD

**Discussion by m4ntr4r4m4** [etsy/open-api#843] [Q&A]
Hello there,
I want to make a small app for creating draft listings.
the main problem I am facing is adding optional parameters like : Primary color, Secondary color, Width, Length.
I read the documentation about createDraftListing and found nothing,
[createDraftListing
](https://developer.etsy.com/documentation/reference#operation/createDraftListing)

your help is appreciated.

**Comment by baahar** [comment]
Hi,

Etsy calls those optional parameters properties and they change depending on the category of a listing. After creating a draft listing you can use the **updateListingProperty** endpoint to add properties that are defined for that listing's category. To find what properties are defined for a category you can use the **getPropertiesByTaxonomyId** endpoint.

Hope this helps.

**Reply by m4ntr4r4m4** [reply]
@baahar  thank you for the information it was helpful.
now I m facing this problem of '' **value_ids** '' when I want to use   **updateListingProperty**  to add the  length property : 
the following example i got it from an active listing.

 "property_id":47626759838,
         "property_name":"Length",
         "scale_id":7,
         "scale_name":"Centimeters",
         "value_ids":[
            59132496576
         ],
         "values":[
            "1"
         ]

how to calculate that **value_ids** ?
and why the scal_name returns null  even when I put "Feet" on request:
{"property_id":47626759838,"property_name":"Length","scale_id":11,**"scale_name":null**,"value_ids":[736473579],"values":["10"]}
{"property_id":47626759898,"property_name":"Width","scale_id":11,"scale_name":null,"value_ids":[946449235],"values":["8"]}

**Reply by munteanulc** [reply]
Hi,

Here's a Python script that you can use as an example:
"""
import requests

url = "https://openapi.etsy.com/v3/application/shops/{yourShopID}/listings"

payload='quantity=3&title=yourTitle&description=yourDescription%0AHow%20about%20that.%0A%0ABla%20Bla%20Bla&price=0.99&who_made=someone_else&when_made=made_to_order&taxonomy_id=1027&shipping_profile_id=yourShippingProfileID&production_partner_ids=yourProductionPartnerID&is_supply=false'
headers = {
  'x-api-key': 'yourAPIKey',
  'Content-Type': 'application/x-www-form-urlencoded',
  'Authorization': 'Bearer yourOAuth20Token'
}

response = requests.request("POST", url, headers=headers, data=payload)

print(response.text)
"""
You can check the API documentation for all the existing properties and for API calls to get their value IDs.
For all the existing nodes, you can use the API and add the query results in a dictionary:
"""
def GetTaxonomyIDNodes():
    import requests
    import json

    url = "https://openapi.etsy.com/v3/application/seller-taxonomy/nodes"
    payload={}
    headers = {
      'x-api-key': 'yourAPIKey',
    }
    response = requests.request("GET", url, headers=headers, data=payload)
    taxonomyTree = json.loads(response.text)
    #print(json.dumps(taxonomyTree, indent = 2))
    taxonomyTreeDictionary = {}
    IDs=[]
    #urlTaxonomyIDs = "https://openapi.etsy.com/v3/application/seller-taxonomy/nodes"
    propertyByTaxonomyID = "https://openapi.etsy.com/v3/application/seller-taxonomy/nodes/1331/properties"
    payload={}
    headers = {
      'x-api-key': 'yourAPIKey',
    }
    response = requests.request("GET", propertyByTaxonomyID, headers=headers, data=payload)
    taxonomyTree = json.loads(response.text)['results']
    print(taxonomyTree)
    
    def getTaxonomyIDs(taxonomyTree):
      for result in taxonomyTree:
        taxonomyTreeDictionary [result['name']] = result['id']
        for child in result['children']:
          if child != '':
            taxonomyTreeDictionary [child['name']] = child['id']
            getTaxonomyIDs(child['children'])
    
    
    getTaxonomyIDs(taxonomyTree)
    with open('rezultate.txt', 'w') as rezultat:
      rezultat.write(json.dumps(taxonomyTreeDictionary, indent=2))
    print(json.dumps(taxonomyTreeDictionary, indent=2))
"""
There is a small issue with the above getRaxonomyIDs function - it does override values if the name of the Key appears several times in the taxonomy id tree. I didn't implement it properly because I don't need it at the moment.

**Reply by m4ntr4r4m4** [reply]
thank you
